From 3a786972c4adec767ba2c2dd5b586cb579d7c7b2 Mon Sep 17 00:00:00 2001
From: Subhendu Sekhar Behera <sbehera@broadcom.com>
Date: Sun, 2 Mar 2014 18:33:53 +0530
Subject: nae: print debug messages using debug level

debug levels:
	DEFAULT		1
	ERROR		1
	FDT		2
	CONFIG		3
	PORT		4
	TRACE		5

default level is 1

usage: modprobe nae nae_debug_level=2
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Nam Ninh <nam.ninh@windriver.com>

diff --git a/drivers/net/ethernet/broadcom/nae/xlpge.h b/drivers/net/ethernet/broadcom/nae/xlpge.h
index 3efb508..e80e75c 100644
--- a/drivers/net/ethernet/broadcom/nae/xlpge.h
+++ b/drivers/net/ethernet/broadcom/nae/xlpge.h
@@ -236,6 +236,25 @@ struct phy_info {
         uint32_t *serdes_addr;
 };
 
+extern int nae_debug_level;
+
+enum nae_dbglevels{
+	NAE_DBG_DEFAULT = 0x1,
+	NAE_DBG_ERROR = 0x1,
+	NAE_DBG_FDT = 0x2,
+	NAE_DBG_CONFIG = 0x4,
+	NAE_DBG_PORT = 0x8,
+	NAE_DBG_GLOBAL = NAE_DBG_PORT | NAE_DBG_CONFIG,
+	NAE_DBG_TRACE = 0x10,
+};
+
+#define nae_print(dbglvl, fmt, args...) \
+{ \
+	if (dbglvl & nae_debug_level) { \
+		printk(fmt, ##args); \
+	} \
+}
+
 #ifdef CONFIG_NLM_ENABLE_LOAD_BALANCING
 /*Flow Meta Info*/
 struct flow_meta_info
@@ -389,18 +408,18 @@ static __inline__ void print_fmn_send_error(const char *str,
 {
 	if(send_result & MSG_DST_FC_FAIL)
 	{
-		printk("[%s] Msg Destination flow control credit fail "
+		nae_print(NAE_DBG_ERROR, "[%s] Msg Destination flow control credit fail "
 			"(send_result=%08x)\n",
 			str, send_result);
 	}
-	else if (send_result & MSG_INFLIGHT_MSG_EX)
-		printk("[%s] MSG_INFLIGHT_MSG_EX(send_result=%08x)\n",
+	else if (send_result & MSG_INFLIGHT_MSG_EX) {
+		nae_print(NAE_DBG_ERROR, "[%s] MSG_INFLIGHT_MSG_EX(send_result=%08x)\n",
 			__func__, send_result);
-	else if (send_result & MSG_TXQ_FULL)
-		printk("[%s] TX message Q full(send_result=%08x)\n",
+	}else if (send_result & MSG_TXQ_FULL) {
+		nae_print(NAE_DBG_ERROR, "[%s] TX message Q full(send_result=%08x)\n",
 			__func__, send_result);
-	else
-		printk("[%s] Unknown send error type(send_result=%08x)\n",
+	}else
+		nae_print(NAE_DBG_ERROR, "[%s] Unknown send error type(send_result=%08x)\n",
 			__func__, send_result);
 }
 
@@ -410,14 +429,14 @@ static __inline__ void dump_buffer(unsigned char *buf, uint32_t len,
 	int k, olen = 0;
 	char out[512];
 
-	printk("\n%s\n", msg);
+	nae_print(NAE_DBG_TRACE, "\n%s\n", msg);
 
 	for (k = 0; k < len; k++) {
 		olen += snprintf(&out[olen], 512 - olen, "<%.2x> ", buf[k]);
 		if ((k + 1) % 16 == 0)
 			olen += snprintf(&out[olen], 512 - olen, "\n");
 	}
-	printk("%s\n", out);
+	nae_print(NAE_DBG_TRACE, "%s\n", out);
 }
 
 static inline int mac_refill_frin_skb(nae_t* nae_cfg, int cpu, uint64_t paddr,
@@ -440,7 +459,7 @@ static inline int mac_refill_frin_skb(nae_t* nae_cfg, int cpu, uint64_t paddr,
 	node_cpu = phys_cpu_to_log_map[node][nae_id][cpu];
 
 	if (nae_cfg == NULL) {
-		printk("%s Error, Invalid node id %d\n", __FUNCTION__, node);
+		nae_print(NAE_DBG_ERROR, "%s Error, Invalid node id %d\n", __FUNCTION__, node);
 		return -1;
 	}
 
@@ -489,7 +508,7 @@ static inline int mac_refill_frin_one_buffer(struct net_device *dev, int cpu,
 
 	skb = nlm_xlp_alloc_skb_atomic(buf_size, priv->node);
 	if (!skb) {
-		printk("[%s] alloc skb failed\n",__FUNCTION__);
+		nae_print(NAE_DBG_ERROR, "[%s] alloc skb failed\n",__FUNCTION__);
 		panic("panic...");
 		return -ENOMEM;
 	}
@@ -501,8 +520,6 @@ static inline int mac_refill_frin_one_buffer(struct net_device *dev, int cpu,
 	return mac_refill_frin_skb(nae_cfg, cpu, (uint64_t)virt_to_bus(skb->data), buf_size, priv->hw_port_id);
 }
 
-
-
 void nlm_xlp_nae_init(void);
 void nlm_xlp_nae_remove(void);
 extern int xlpge_eeprom_init(void);
diff --git a/drivers/net/ethernet/broadcom/nae/xlpge_ethtool.c b/drivers/net/ethernet/broadcom/nae/xlpge_ethtool.c
index 1194e3d..3cb659b 100644
--- a/drivers/net/ethernet/broadcom/nae/xlpge_ethtool.c
+++ b/drivers/net/ethernet/broadcom/nae/xlpge_ethtool.c
@@ -316,7 +316,7 @@ static void xlp_get_strings (struct net_device *dev, u32 stringset, u8 *buf)
 			sizeof(xlp_ethtool_stats_keys));
 		break;
 	default:
-		printk(KERN_WARNING "%s: Invalid stringset %d\n",
+		nae_print(NAE_DBG_ERROR, "%s: Invalid stringset %d\n",
 			__func__, stringset);
 		break;
 	}
diff --git a/drivers/net/ethernet/broadcom/nae/xlpge_lro.h b/drivers/net/ethernet/broadcom/nae/xlpge_lro.h
index 11e1651..4bf1172 100644
--- a/drivers/net/ethernet/broadcom/nae/xlpge_lro.h
+++ b/drivers/net/ethernet/broadcom/nae/xlpge_lro.h
@@ -81,7 +81,7 @@ static void lro_init(struct net_device *dev)
 	priv = netdev_priv(dev);
 
 	if (enable_lro) {
-		printk("LRO is enabled \n");
+		nae_print(NAE_DBG_CONFIG, "LRO is enabled \n");
 		dev->features |= NETIF_F_LRO;
 	//	dev->features |= NETIF_F_GRO;
 	//	return;
diff --git a/drivers/net/ethernet/broadcom/nae/xlpge_nae.c b/drivers/net/ethernet/broadcom/nae/xlpge_nae.c
index 8e158c0..39096d4 100644
--- a/drivers/net/ethernet/broadcom/nae/xlpge_nae.c
+++ b/drivers/net/ethernet/broadcom/nae/xlpge_nae.c
@@ -116,6 +116,12 @@ uint32_t last_rcvd_node[NR_CPUS * 8] ____cacheline_aligned;
 uint64_t last_rcvd_skb_phys[NR_CPUS * 8] ____cacheline_aligned;
 #endif
 
+int nae_debug_level = 1;
+module_param(nae_debug_level, int, 0);
+MODULE_PARM_DESC(nae_debug_level,
+	"Vaild debug levels 1-5 (default="
+		__MODULE_STRING(nae_debug_level)")");
+
 static unsigned short nlm_select_queue(struct net_device *dev,
 				       struct sk_buff *skb)
 {
@@ -139,7 +145,7 @@ static int init_dummy_entries_for_port_fifos(nae_t* nae_cfg)
 
 	skb = nlm_xlp_alloc_skb_atomic(size, node);
 	if (!skb) {
-		printk("[%s] alloc skb failed\n", __func__);
+		nae_print(NAE_DBG_ERROR, "[%s] alloc skb failed\n", __func__);
 		panic("panic...");
 		return -ENOMEM;
 	}
@@ -154,7 +160,7 @@ static int init_dummy_entries_for_port_fifos(nae_t* nae_cfg)
 	return 0;
 
 //	msgrng_access_enable(mflags);
-	printk("Total free ins = 0x%x\n", nae_cfg->frin_total_queue);
+	nae_print(NAE_DBG_CONFIG, "Total free ins = 0x%x\n", nae_cfg->frin_total_queue);
 	for (i = 0; i < nae_cfg->frin_total_queue; i++) {
 		/* nothing to do, if it is owned by some domain */
 		if((1 << i) & fifo_mask)
@@ -169,20 +175,20 @@ static int init_dummy_entries_for_port_fifos(nae_t* nae_cfg)
 		for (j = 0; j < 4; j++) {
 			//netsoc_nae_send_freein_buf(nae_cfg, vc_index, msg);
 //#if 0
-				printk("Sending message check freein carving (qid=%d)\n",
+				nae_print(NAE_DBG_TRACE, "Sending message check freein carving (qid=%d)\n",
 					vc_index);
 			if ((ret = xlp_message_send_1(vc_index, code, msg))
 				& 0x7) {
 				print_fmn_send_error(__func__, ret);
-				printk("Unable to send configured free desc\n");
-				printk("check freein carving (qid=%d)\n",
+				nae_print(NAE_DBG_ERROR, "Unable to send configured free desc\n");
+				nae_print(NAE_DBG_TRACE, "check freein carving (qid=%d)\n",
 					vc_index);
 				rv = -1;
 				goto err;
 			}
 //#endif
 		}
-		printk("Send %d dummy descriptors for queue %d(vc %d) of length %d\n",
+		nae_print(NAE_DBG_TRACE, "Send %d dummy descriptors for queue %d(vc %d) of length %d\n",
 			j, i, vc_index, size);
 	}
 err:
@@ -265,7 +271,7 @@ static int nlm_configure_shared_freein_fifo(int node,
 	int len = 0, i = 0;
 	char *paddr_info, *desc_info;
 
-	printk("%s in \n", __FUNCTION__);
+	nae_print(NAE_DBG_TRACE, "%s in \n", __FUNCTION__);
 
 	for(shdom = 0; shdom <= NLM_NAE_MAX_SHARED_DOMS; shdom++) {
 		if(!nae_cfg->shinfo[shdom].valid)
@@ -280,7 +286,7 @@ static int nlm_configure_shared_freein_fifo(int node,
 		desc_info = nae_cfg->shinfo[shdom].desc_info;
 		desc_info_len = nae_cfg->shinfo[shdom].desc_info_len;
 
-		printk("shind %d dom %d repl %d paddr_ptr %lx len %d desc_ptr %lx dlen %d\n",
+		nae_print(NAE_DBG_TRACE, "shind %d dom %d repl %d paddr_ptr %lx len %d desc_ptr %lx dlen %d\n",
 				shdom, nae_cfg->shinfo[shdom].domid, owner_replenish,
 				(long)paddr_info, paddr_info_len, (long)desc_info, desc_info_len);
 
@@ -303,7 +309,7 @@ static int nlm_configure_shared_freein_fifo(int node,
 				break;
 		} while(len < paddr_info_len);
 
-		printk("domid %d node %d addr %llx size %llx\n",
+		nae_print(NAE_DBG_TRACE, "domid %d node %d addr %llx size %llx\n",
 					nae_cfg->shinfo[shdom].domid, cnode,
 					paddr, psize);
 
@@ -325,7 +331,7 @@ static int nlm_configure_shared_freein_fifo(int node,
 			if(cnode != node)
 				continue;
 
-			printk("node %d fmask %x dsize %d dppadsz %d ndescs %d \n",
+			nae_print(NAE_DBG_TRACE, "node %d fmask %x dsize %d dppadsz %d ndescs %d \n",
 					cnode, fmask, dsize, dppadsz, ndescs);
 
 			for(fifo = 0; fifo < NLM_NAE_MAX_FREEIN_FIFOS_PER_NODE;
@@ -337,7 +343,7 @@ static int nlm_configure_shared_freein_fifo(int node,
 				for(descs = 0; descs < ndescs; descs++) {
 					if((paddr + dsize) > epaddr) {
 						msgrng_access_disable(mflags);
-						printk("Error, descriptors \
+						nae_print(NAE_DBG_ERROR, "Error, descriptors \
 							buffer overflow \n");
 						err = -1;
 						goto err_exit;
@@ -348,7 +354,7 @@ static int nlm_configure_shared_freein_fifo(int node,
 					rv = xlp_message_send_1(vc_index, code, msg);
 					if(rv & 0x7) {
 						msgrng_access_disable(mflags);
-						printk("Unable to send \
+						nae_print(NAE_DBG_ERROR, "Unable to send \
 							configured free desc, \
 							check freein carving \
 							(qid=%d)\n", vc_index);
@@ -360,7 +366,7 @@ static int nlm_configure_shared_freein_fifo(int node,
 
 				}
 				msgrng_access_disable(mflags);
-				printk("Send %d descriptors for queue \
+				nae_print(NAE_DBG_TRACE, "Send %d descriptors for queue \
 						%d(%d) of length %d\n",
 						ndescs, fifo, vc_index, dsize);
 
@@ -397,7 +403,7 @@ static int initialize_nae_per_node(nae_t * nae_cfg, uint32_t *phys_cpu_map, int
 
 
 	for (i = 0; i <= NLM_NAE_MAX_SHARED_DOMS; i++) {
-		printk("naeid %d shind %d dom %d valid %x numcpus %d\n", nae_id,
+		nae_print(NAE_DBG_CONFIG, "naeid %d shind %d dom %d valid %x numcpus %d\n", nae_id,
 				i, nae_cfg->shinfo[i].domid, nae_cfg->shinfo[i].valid, nae_cfg->shinfo[i].num_cpus);
 		lnx_shinfo[i] = (struct nlm_nae_linux_shinfo *)mem;
 		lnx_shinfo[i]->flags  = nae_cfg->shinfo[i].valid ? NLM_NAE_LNX_SHINFO_FL_VALID : 0;
@@ -409,7 +415,7 @@ static int initialize_nae_per_node(nae_t * nae_cfg, uint32_t *phys_cpu_map, int
 		for(cpus = 0; cpus < lnx_shinfo[i]->num_cpus; cpus++)  {
 			int phys_cpu;
 			lnx_shinfo[i]->fwd_info[cpus] = nae_cfg->shinfo[i].fwd_info[cpus];
-			printk("cpu %d value %08x\n", cpus, nae_cfg->shinfo[i].fwd_info[cpus]);
+			nae_print(NAE_DBG_TRACE, "cpu %d value %08x\n", cpus, nae_cfg->shinfo[i].fwd_info[cpus]);
 			phys_cpu = (lnx_shinfo[i]->fwd_info[cpus] >> NLM_NAE_LNX_SHINFO_FWD_INFO_LCPU_2_PCPU_OFF) &
 				NLM_NAE_LNX_SHINFO_FWD_INFO_LCPU_2_PCPU_MASK;
 			phys_cpu_to_log_map[node][nae_id][phys_cpu] = cpus;
@@ -470,8 +476,8 @@ static int initialize_nae_per_node(nae_t * nae_cfg, uint32_t *phys_cpu_map, int
 					(cpu_2_jumbo_frfifo[node][nae_id][cpus] << NLM_NAE_LNX_SHINFO_FWD_INFO_LCPU_2_JF_OFF);
 			}
 		} else {
-			printk("freein-fifo unavailable: ");
-			printk("Disabling Jumbo\n");
+			nae_print(NAE_DBG_DEFAULT, "freein-fifo unavailable: ");
+			nae_print(NAE_DBG_DEFAULT, "Disabling Jumbo\n");
 			*jumbo_enabled = 0;
 		}
 	} else if (*jumbo_enabled == 0) {
@@ -496,9 +502,9 @@ static int initialize_nae_per_node(nae_t * nae_cfg, uint32_t *phys_cpu_map, int
 	if (nae_cfg->owned) {
 		if(netsoc_write_ucore_shmem(nae_cfg,
 			(uint32_t *)lnx_shinfo[0],  size/sizeof(uint32_t)) != 0) {
-			printk("Error, Write ucore sram failed!!!, size %d\n", size);
+			nae_print(NAE_DBG_ERROR, "Error, Write ucore sram failed!!!, size %d\n", size);
 		} else
-			printk("Write ucore sram success, size %d\n", size);
+			nae_print(NAE_DBG_TRACE, "Write ucore sram success, size %d\n", size);
 
 		netsoc_restart_ucore_using_fdt(nae_cfg, fdt);
 	}
@@ -539,10 +545,10 @@ static int initialize_nae_per_node(nae_t * nae_cfg, uint32_t *phys_cpu_map, int
 
 #if 1
 	for (i = 0; i <= NLM_NAE_MAX_SHARED_DOMS; i++) {
-		printk("naeid %d domid %d node %d flag %x\n", nae_id,
+		nae_print(NAE_DBG_CONFIG, "naeid %d domid %d node %d flag %x\n", nae_id,
 				lnx_shinfo[i]->domid, lnx_shinfo[i]->node, lnx_shinfo[i]->flags);
 		for(cpus = 0; cpus < lnx_shinfo[i]->num_cpus; cpus++) {
-			printk("lcpu %d value %08x (pcpu %d rxfifo %d jfifo %d)\n",
+			nae_print(NAE_DBG_TRACE, "lcpu %d value %08x (pcpu %d rxfifo %d jfifo %d)\n",
 					cpus, lnx_shinfo[i]->fwd_info[cpus],
 					(lnx_shinfo[i]->fwd_info[cpus] >> NLM_NAE_LNX_SHINFO_FWD_INFO_LCPU_2_PCPU_OFF) & 0xff ,
 					(lnx_shinfo[i]->fwd_info[cpus] >> NLM_NAE_LNX_SHINFO_FWD_INFO_LCPU_2_RF_OFF) & 0xff,
@@ -612,14 +618,15 @@ int initialize_nae(uint32_t *phys_cpu_map, int mode, int *jumbo_enabled)
 	mod_api.writel = NULL;
 	mod_api.free = NULL;
 	mod_api.contig_free = NULL;
+	mod_api.netsoc_debug_level = nae_debug_level;
 
 	brcm_netsoc_lib_init(&mod_api);
 
 	if (init_netsoc(fdt, dom_id, &mod_api) < 0) {
-                printk("NETSOC initialization failed \n");
+                nae_print(NAE_DBG_ERROR, "NETSOC initialization failed \n");
 		return -1;
 	}
-	printk("DONE WITH INIT NETSOC #######\n");
+	nae_print(NAE_DBG_DEFAULT, "DONE WITH INIT NETSOC #######\n");
 
 	/*get max nae*/
 	max_nae_units = get_num_nae_pernode();
@@ -632,7 +639,7 @@ int initialize_nae(uint32_t *phys_cpu_map, int mode, int *jumbo_enabled)
 	}
 
 	msgrng_access_disable(mflags);
-	printk("%s done\n", __FUNCTION__);
+	nae_print(NAE_DBG_DEFAULT, "%s done\n", __FUNCTION__);
 	return 0;
 }
 
@@ -663,7 +670,7 @@ static int nlm_replenish_per_cpu_buffer(nae_t* nae_cfg,
 		vc_index = qindex + nae_cfg->frin_queue_base;
 		skb = nlm_xlp_alloc_skb_atomic(size, node);
 		if (!skb) {
-			printk("[%s] alloc skb failed\n",__FUNCTION__);
+			nae_print(NAE_DBG_ERROR, "[%s] alloc skb failed\n",__FUNCTION__);
 			break;
 		}
 
@@ -681,8 +688,8 @@ static int nlm_replenish_per_cpu_buffer(nae_t* nae_cfg,
 
 		if ((ret = xlp_message_send_1(vc_index, code, msg)) & 0x7) {
 			print_fmn_send_error(__func__, ret);
-			printk("Unable to send configured free desc, ");
-			printk("check freein carving (qid=%d)\n", vc_index);
+			nae_print(NAE_DBG_ERROR, "Unable to send configured free desc, ");
+			nae_print(NAE_DBG_ERROR, "check freein carving (qid=%d)\n", vc_index);
 
 			/* free the buffer and return! */
 			dev_kfree_skb_any(skb);
@@ -694,7 +701,7 @@ static int nlm_replenish_per_cpu_buffer(nae_t* nae_cfg,
 		msgrng_access_disable(mflags);
 	}
 
-	printk("Send %d descriptors for queue %d(vc %d) of length %d\n",
+	nae_print(NAE_DBG_TRACE, "Send %d descriptors for queue %d(vc %d) of length %d\n",
 		bufcnt, qindex, vc_index, size);
 
 	return ret;
@@ -899,7 +906,7 @@ static int p2p_desc_mem_init(void)
 			CACHELINE_SIZE - 1) & (~((CACHELINE_SIZE)-1)));
 	tsize = dsize * MAX_TSO_SKB_PEND_REQS;
 
-	printk("%s in, dsize %d tsize %d \n", __FUNCTION__, dsize, tsize);
+	nae_print(NAE_DBG_CONFIG, "%s in, dsize %d tsize %d \n", __FUNCTION__, dsize, tsize);
 
 	for (cpu = 0; cpu < NR_CPUS; cpu++) {
 		buf = cacheline_aligned_kmalloc(tsize, GFP_KERNEL);
@@ -919,7 +926,7 @@ static int p2p_desc_mem_init(void)
 
 static void nlm_enable_l3_l4_parser(nae_t *nae)
 {
-	printk("Enabling parser for nae node %d naeid %d\n",
+	nae_print(NAE_DBG_TRACE, "Enabling parser for nae node %d naeid %d\n",
 			nae->node, nae->nae_id);
 	l2_parser_config_t l2;
 	l3_parser_config_t l3;
@@ -1033,7 +1040,7 @@ static int  nlm_xlp_nae_open (struct net_device *dev)
 		if (request_irq( irq, nlm_xlp_nae_int_handler,
 			IRQF_SHARED, dev->name, dev)) {
 			ret = -EBUSY;
-			printk("can't get mac interrupt line (%d)\n",dev->irq);
+			nae_print(NAE_DBG_ERROR, "can't get mac interrupt line (%d)\n",dev->irq);
 		}
 		dump_irt_entry(PIC_IRT_NA_INDEX(port));
 	}
@@ -1183,10 +1190,10 @@ static void xlp_mac_setup_hwaddr(struct dev_data *priv)
 static int  nlm_xlp_nae_ioctl (struct net_device *dev, struct ifreq *rq, int cmd)
 {
 	int rc = 0;
-	printk("nlm_xlp_nae_ioctl called \n");
+	nae_print(NAE_DBG_TRACE, "nlm_xlp_nae_ioctl called \n");
 	switch (cmd) {
 	case SIOCSHWTSTAMP:
-		printk("HW time stamping supported by HW\n");
+		nae_print(NAE_DBG_CONFIG, "HW time stamping supported by HW\n");
 		return 0;
 	default:
 		rc = -EOPNOTSUPP;
@@ -1211,14 +1218,14 @@ static int nlm_xlp_nae_change_mtu(struct net_device *dev, int new_mtu)
 
 	if (enable_jumbo &&
 		(new_mtu > ETH_JUMBO_DATA_LEN || new_mtu < ETH_ZLEN)) {
-		printk ("MTU should be between %d and %d\n",
+		nae_print(NAE_DBG_TRACE, "MTU should be between %d and %d\n",
 			ETH_ZLEN, ETH_JUMBO_DATA_LEN);
 		return -EINVAL;
 	}
 
 	if (!enable_jumbo &&
 		(new_mtu > ETH_DATA_LEN || new_mtu < ETH_ZLEN)) {
-		printk ("MTU should be between %d and %d\n",
+		nae_print(NAE_DBG_TRACE, "MTU should be between %d and %d\n",
 			ETH_ZLEN, ETH_DATA_LEN);
 		return -EINVAL;
 	}
@@ -1299,7 +1306,7 @@ static void  nlm_xlp_nae_tx_timeout (struct net_device *dev)
 
 	netif_tx_wake_all_queues(dev);
 
-	printk(KERN_WARNING "%s: Transmit timed out\n", dev->name);
+	nae_print(NAE_DBG_ERROR, "%s: Transmit timed out\n", dev->name);
 	return;
 }
 #if 0
@@ -1350,7 +1357,7 @@ static irqreturn_t nlm_xlp_nae_int_handler(int irq, void *dev_id)
 	priv->cpu_stats[cpu].interrupts++;
 
 	if(!dev_id) {
-		printk("[%s]: NULL dev_id \n", __FUNCTION__ );
+		nae_print(NAE_DBG_ERROR, "[%s]: NULL dev_id \n", __FUNCTION__ );
 		return IRQ_HANDLED;
 	}
 	dev = (struct net_device*)dev_id;
@@ -1437,11 +1444,11 @@ static int nlm_per_port_nae_init(nae_t* nae_cfg, int port, int maxnae)
 		priv->phy.addr = nae_cfg->ports[port].ext_phy_addr;
 		if (nae_cfg->ports[port].hw_port_id == 0) {
 			if (dev_alloc_name(dev, "ilk0-%d") < 0)
-				printk("alloc name failed \n");
+				nae_print(NAE_DBG_ERROR, "alloc name failed \n");
 		}
 		else {
 			if (dev_alloc_name(dev, "ilk8-%d") < 0)
-				printk("alloc name failed \n");
+				nae_print(NAE_DBG_ERROR, "alloc name failed \n");
 		}
 		break;
 	default:
@@ -1488,15 +1495,18 @@ void nlm_xlp_nae_init(void)
 	unsigned char *mode_str[3] = {"INVALID","TCP_PERF","ROUTE_PERF"};
 	nae_t* nae_cfg;
 
+	nae_debug_level = (1 << nae_debug_level) - 1;
+
 	if (!(perf_mode == NLM_TCP_MODE || perf_mode == NLM_RT_MODE)) {
-		printk("Invalid perf mode passed -- Using TCP_PERF mode\n");
+		nae_print(NAE_DBG_DEFAULT, "Invalid perf mode passed -- Using TCP_PERF mode\n");
 		perf_mode = NLM_TCP_MODE;
 	}
 
-	printk("======= Module Parameters =========\n");
-	printk("num_descs_per_normalq=%d num_descs_per_jumboq=%d ",
+	nae_print(NAE_DBG_DEFAULT, "======= Module Parameters =========\n");
+	nae_print(NAE_DBG_DEFAULT, "nae debug mask 0x%x\n", nae_debug_level); 
+	nae_print(NAE_DBG_DEFAULT, "num_descs_per_normalq=%d num_descs_per_jumboq=%d ",
 	       num_descs_per_normalq, num_descs_per_jumboq);
-	printk("perf_mode=%s enable_lro=%d enable_jumbo=%d \n",
+	nae_print(NAE_DBG_DEFAULT, "perf_mode=%s enable_lro=%d enable_jumbo=%d \n",
 	       mode_str[perf_mode], enable_lro, enable_jumbo);
 
 	for (i = 0; i < NR_CPUS; i++)
@@ -1537,7 +1547,7 @@ void nlm_xlp_nae_init(void)
 				       xlp_mac_proc_read,	/* proc fn */
 				       0	/* no client data */);
 	if (!entry) {
-		printk("[%s]: Unable to create proc entry for xlp_mac!\n",
+		nae_print(NAE_DBG_ERROR, "[%s]: Unable to create proc entry for xlp_mac!\n",
 		       __func__);
 	}
 	entry = create_proc_read_entry("nae_stat",
@@ -1546,7 +1556,7 @@ void nlm_xlp_nae_init(void)
 				       nae_proc_read,
 				       0);
 	if (!entry) {
-		printk("[%s]: Unable to create proc entry for nae_proc!\n",
+		nae_print(NAE_DBG_ERROR,"[%s]: Unable to create proc entry for nae_proc!\n",
 		       __func__);
 	}
 
@@ -1568,16 +1578,16 @@ void nlm_xlp_nae_init(void)
 				);
 		if (entry)
 			entry->proc_fops = &nlm_load_balance_proc_fops;
-		printk("Enabling load balance option\n");
+		nae_print(NAE_DBG_DEFAULT, "Enabling load balance option\n");
 		nlm_init_load_balance();
 	} else {
-		printk("Disabling load balance option\n");
+		nae_print(NAE_DBG_DEFAULT, "Disabling load balance option\n");
 		load_balance_en = 0;
 	}
 #endif
 
 	if (replenish_freein_fifos() != 0)
-		printk("Replenishmemt of freein fifos failed\n");
+		nae_print(NAE_DBG_ERROR, "Replenishmemt of freein fifos failed\n");
 
 	if (enable_napi)
 		nlm_xlp_enable_napi();
diff --git a/drivers/net/ethernet/broadcom/nae/xlpge_proc.c b/drivers/net/ethernet/broadcom/nae/xlpge_proc.c
index 157b5a3..bdcd8b2 100644
--- a/drivers/net/ethernet/broadcom/nae/xlpge_proc.c
+++ b/drivers/net/ethernet/broadcom/nae/xlpge_proc.c
@@ -56,7 +56,7 @@ int nae_proc_read(char *page, char **start, off_t off,
 				(err_replenish_count[CPU_INDEX(i)] == 0))
 			continue;
 
-		printk("cpu%d, recv %ld fast_repl %ld, slow_repl %ld "
+		nae_print(NAE_DBG_TRACE, "cpu%d, recv %ld fast_repl %ld, slow_repl %ld "
 			"err_repl %ld p2pdalloc %lld\n", i,
 			(ulong)receive_count[CPU_INDEX(i)],
 			(ulong)fast_replenish_count[CPU_INDEX(i)],
@@ -76,7 +76,7 @@ int nae_proc_read(char *page, char **start, off_t off,
 		receive_count[CPU_INDEX(i)] = 0;
 	}
 	/*check how many hash are empty...*/
-	printk("TOTAL_FAST_REPL %ld, TOTAL_SLOW_REPL %ld,"
+	nae_print(NAE_DBG_CONFIG, "TOTAL_FAST_REPL %ld, TOTAL_SLOW_REPL %ld,"
 	       " TOTAL_ERR_REPL %ld TOTAL_RECV %ld\n",
 			(ulong)total_fast,
 			(ulong)total_slow,
diff --git a/drivers/net/ethernet/broadcom/nae/xlpge_rx.c b/drivers/net/ethernet/broadcom/nae/xlpge_rx.c
index 1d54eb6..7ce9628 100644
--- a/drivers/net/ethernet/broadcom/nae/xlpge_rx.c
+++ b/drivers/net/ethernet/broadcom/nae/xlpge_rx.c
@@ -107,7 +107,7 @@ inline void process_tx_complete(int cpu, uint32_t src_id, uint64_t msg0)
                 dev_kfree_skb_any(skb);
         }
         else {
-                printk("[%s]: [txc] Null skb? "
+                nae_print(NAE_DBG_ERROR, "[%s]: [txc] Null skb? "
                        " paddr = %llx (halting cpu!)\n", __func__, addr);
                 cpu_halt();
         }
@@ -176,7 +176,7 @@ static void nlm_remove_inactive_flow(int cpu)
 		fmi = nlm_flow_meta_info + index;
 #ifdef LOAD_BALANCE_DEBUG_ENABLE
 		if (index > NLM_UCORE_SHARED_TABLE_SIZE)
-			printk("??? Index %d\n",index);
+			nae_print(NAE_DBG_TRACE, "??? Index %d\n",index);
 #endif
 		bytes_received = fmi->total_bytes_rcvd - fmi->last_sample_bytes;
 		fmi->last_sample_bytes = fmi->total_bytes_rcvd;
@@ -184,7 +184,7 @@ static void nlm_remove_inactive_flow(int cpu)
 		if (bytes_received == 0) {
 #ifdef LOAD_BALANCE_DEBUG_ENABLE
 			if (fmi->cpu_owner != cpu) {
-				printk("Error!! fmi->cpu_owner = %lld, cpu %d, "
+				nae_print(NAE_DBG_ERROR, "Error!! fmi->cpu_owner = %lld, cpu %d, "
 					"entry %d, total_active_flows %lld\n",
 					fmi->cpu_owner, cpu, i,
 					afl->nr_active_flows);
@@ -235,7 +235,7 @@ static void setup_search_path(void)
 		num_phy_cpu++;
 	}
 #ifdef LOAD_BALANCE_DEBUG_ENABLE
-	printk("num_phy_cpu %d, nlm_pcpu_mask %#x\n",
+	nae_print(NAE_DBG_TRACE, "num_phy_cpu %d, nlm_pcpu_mask %#x\n",
 		num_phy_cpu, nlm_pcpu_mask[0]);
 #endif
 	for (pcpu = 0; pcpu < NUM_LOAD_BALANCE_CPU; pcpu++) {
@@ -296,7 +296,7 @@ static void setup_search_path(void)
 		}
 	}
 #ifdef LOAD_BALANCE_DEBUG_ENABLE
-	printk("\nSearchPath:\n");
+	nae_print(NAE_DBG_TRACE, "\nSearchPath:\n");
 	for (i = 0; i < NUM_LOAD_BALANCE_CPU; i++) {
 		int len;
 		sprintf(buf, "Cpu%d ==> ", i);
@@ -307,7 +307,7 @@ static void setup_search_path(void)
 			len = strlen(buf);
 		}
 		sprintf(buf + len,"\n");
-		printk(buf);
+		nae_print(NAE_DBG_TRACE, (buf);
 	}
 #endif
 	return;
@@ -439,14 +439,14 @@ void nlm_init_load_balance(void)
 		NUM_LOAD_BALANCE_CPU);
 
 	if (!nlm_active_flow_list) {
-	printk("\nAllocation Failed!!! for size %lu bytes\n",
+	nae_print(NAE_DBG_ERROR, "Allocation Failed!!! for size %lu bytes\n",
 		sizeof(struct active_flow_list) * NUM_LOAD_BALANCE_CPU);
 		return;
 	}
 	memset(nlm_active_flow_list, 0, sizeof(struct active_flow_list) *
 		NUM_LOAD_BALANCE_CPU);
 #ifdef LOAD_BALANCE_DEBUG_ENABLE
-	printk("Allocated active_flow_list @ %#lx, size %lu bytes\n",
+	nae_print(NAE_DBG_CONFIG, "Allocated active_flow_list @ %#lx, size %lu bytes\n",
 		(unsigned long)nlm_active_flow_list,
 		sizeof(struct active_flow_list) * NUM_LOAD_BALANCE_CPU);
 #endif 
@@ -454,14 +454,14 @@ void nlm_init_load_balance(void)
 		(NLM_UCORE_SHARED_TABLE_SIZE + 1));
 
 	if (!nlm_flow_meta_info) {
-		printk("\nAllocation Failed!!! for size %lu bytes\n",
+		nae_print(NAE_DBG_ERROR, "\nAllocation Failed!!! for size %lu bytes\n",
 			sizeof(struct flow_meta_info) * NUM_LOAD_BALANCE_CPU);
 		return;
 	}
 	memset(nlm_flow_meta_info, 0, sizeof(struct flow_meta_info) *
 		(NLM_UCORE_SHARED_TABLE_SIZE + 1));
 #ifdef LOAD_BALANCE_DEBUG_ENABLE
-	printk("Allocated flow_meta_info @ %#lx, size %lu bytes\n",
+	nae_print(NAE_DBG_CONFIG, "Allocated flow_meta_info @ %#lx, size %lu bytes\n",
 		(ulong)nlm_flow_meta_info, sizeof(struct flow_meta_info) *
 		(NLM_UCORE_SHARED_TABLE_SIZE + 1));
 #endif 
@@ -488,7 +488,7 @@ void nlm_init_load_balance(void)
 	ucore_shared_data = vmalloc(NLM_UCORE_SHARED_TABLE_SIZE *
 		sizeof(uint32_t));
 	if (!ucore_shared_data) {
-		printk("Ucore updation failed!!\n");
+		nae_print(NAE_DBG_ERROR, "Ucore updation failed!!\n");
 		return;
 	}
 
@@ -519,7 +519,7 @@ static void dump_packet(unsigned char *vaddr, int len)
 {
 	int i = 0;
 	for (i = 0; i < len; i+=8) {
-		printk("[%#x] [%#x] [%#x] [%#x] [%#x] [%#x] [%#x] [%#x]\n",
+		nae_print(NAE_DBG_TRACE, "[%#x] [%#x] [%#x] [%#x] [%#x] [%#x] [%#x] [%#x]\n",
 			*(vaddr + i + 0), *(vaddr + i + 1), *(vaddr + i + 2),
 			*(vaddr + i + 3), *(vaddr + i + 4), *(vaddr + i + 5),
 			*(vaddr + i + 6), *(vaddr + i + 7));
@@ -533,11 +533,11 @@ static void dump_prepad(unsigned char *vaddr)
 
 	if (*tmp == 0xdeadbeef) {
 		for(i = 0; i < 4; i++)
-			printk("[%d] ==> [%#x]\n", i, *(tmp + i));
+			nae_print(NAE_DBG_TRACE, "[%d] ==> [%#x]\n", i, *(tmp + i));
 	} else {
 #if 1
 	for (i = 0; i < 16; i+=8){
-		printk("[%#x] [%#x] [%#x] [%#x] [%#x] [%#x] [%#x] [%#x]\n",
+		nae_print(NAE_DBG_TRACE, "[%#x] [%#x] [%#x] [%#x] [%#x] [%#x] [%#x] [%#x]\n",
 			*(vaddr + i + 0), *(vaddr + i + 1), *(vaddr + i + 2),
 			*(vaddr + i + 3), *(vaddr + i + 4), *(vaddr + i + 5),
 			*(vaddr + i + 6), *(vaddr + i + 7));
@@ -563,7 +563,7 @@ static inline void nlm_update_flow_stats(unsigned int *prepad,
 	if (*prepad != NLM_LOAD_BALANCING_MAGIC) {
 		/*No extractions have happend...*/
 #ifdef LOAD_BALANCE_DEBUG_ENABLE
-		printk("Invalid Packet!!\n");
+		nae_print(NAE_DBG_ERROR, "Invalid Packet!!\n");
 		dump_packet(prepad, 64 + 16);
 #endif
 		return;
@@ -588,7 +588,7 @@ static inline void nlm_update_flow_stats(unsigned int *prepad,
 		index = afl->nr_active_flows;
 #ifdef LOAD_BALANCE_DEBUG_ENABLE
 		if (index >= NLM_UCORE_SHARED_TABLE_SIZE)
-			printk("Index %lld Crossing Table size...\n", index);
+			nae_print(NAE_DBG_TRACE, "Index %lld Crossing Table size...\n", index);
 #endif
 		afl->index_to_flow_meta_info[index] = hash_index;
 		mb();
@@ -618,7 +618,7 @@ static int inline valid_buffer_lifo(int cpu, uint64_t msg1, uint32_t src_id)
 		if(is_nlm_xlp9xx()){
 				nae = get_nae(node, nae_id);
 				if(!(nae->freein_fifo_dom_mask & (1<<lifo))){
-						printk("Error Packet: Cpu %d, msg1 %#lx, lifo %d, Owner fifo mask %#lx\n", 
+						nae_print(NAE_DBG_ERROR, "Error Packet: Cpu %d, msg1 %#lx, lifo %d, Owner fifo mask %#lx\n", 
 										cpu, msg1, lifo, nae->freein_fifo_dom_mask);
 						msgrng_access_enable(mflags);
 						qid = nae->frin_queue_base + lifo;
@@ -673,7 +673,7 @@ static inline void process_rx_packets(void *arg, int cpu, unsigned int src_id,
 
 #ifdef ENABLE_SANITY_CHECKS
 	if (port >= MAX_GMAC_PORT) {
-		printk("[%s]: bad port=%d, context=%d\n", __func__,
+		nae_print(NAE_DBG_ERROR, "[%s]: bad port=%d, context=%d\n", __func__,
 			port, context);
 		/*TODO: Where to replenish this packet ???? Context is out of range!*/
 		return;
@@ -683,13 +683,13 @@ static inline void process_rx_packets(void *arg, int cpu, unsigned int src_id,
 
 #ifdef ENABLE_SANITY_CHECKS
 	if(nae_cfg->nae_id!=0 && nae_cfg->nae_id !=1)
-		printk("Nae ID is  wrong\n");
+		nae_print(NAE_DBG_ERROR, "Nae ID is  wrong\n");
 #endif
 	pdev = per_cpu_netdev[node][cpu][nae_cfg->nae_id][port];
 	 
 #ifdef ENABLE_SANITY_CHECKS
 	if (!pdev) {
-		printk("[%s]: [rx] wrong port=%d(context=%d)? pdev = NULL!\n",
+		nae_print(NAE_DBG_ERROR, "[%s]: [rx] wrong port=%d(context=%d)? pdev = NULL!\n",
 			__func__, port, context);
 		return;
 	}
@@ -710,7 +710,7 @@ static inline void process_rx_packets(void *arg, int cpu, unsigned int src_id,
 
 #ifdef ENABLE_SANITY_CHECKS
 	if (!skb) {
-		printk("[%s] Null skb? addr=%llx, vaddr=%llx, "
+		nae_print(NAE_DBG_TRACE, "[%s] Null skb? addr=%llx, vaddr=%llx, "
 		       "dropping it and losing one buffer!\n",
 				__func__, addr, vaddr);
 		err_replenish_count[CPU_INDEX(cpu)]++;
@@ -734,11 +734,11 @@ static inline void process_rx_packets(void *arg, int cpu, unsigned int src_id,
 	nae_cfg = priv->nae;
 
 #ifdef MACSEC_DEBUG
-	printk("%s nae_cfg->sectag_offset = %d sectag_len = %d icv_len = %d\n",
+	nae_print(NAE_DBG_CONFIG, "%s nae_cfg->sectag_offset = %d sectag_len = %d icv_len = %d\n",
 		__func__, nae_cfg->sectag_offset[port],
 		nae_cfg->sectag_len[port], nae_cfg->icv_len[port]);
 	dump_buffer(skb->data, len, "RX skb pkt:");
-	printk("msec_port = %x port = %d len = %d \
+	nae_print(NAE_DBG_CONFIG, "msec_port = %x port = %d len = %d \
 		nae_cfg->msec_rx_port_enable = %x\n",msec_port, port,
 		len, nae_cfg->msec_rx_port_enable);
 #endif
@@ -884,7 +884,7 @@ static int xlp_poll_lower(void *arg, int budget, int cpu)
 		no_rx_pkt_rcvd++;
 #ifdef ENABLE_SANITY_CHECKS
 		if ((size < 1) && (size > 3)) {
-			printk("Unexpected single entry packet\n");
+			nae_print(NAE_DBG_ERROR, "Unexpected single entry packet\n");
 			continue;
 		}
 #endif
@@ -893,7 +893,7 @@ static int xlp_poll_lower(void *arg, int budget, int cpu)
 		else if (size == 1)
 			process_tx_complete(cpu, src_id, msg0);
 		else {
-			printk("%s , Error invalid message, size %d\n",
+			nae_print(NAE_DBG_ERROR, "%s , Error invalid message, size %d\n",
 				__func__, size);
 			continue;
 		}	
@@ -925,7 +925,7 @@ static void nlm_xlp_nae_msgring_handler(uint32_t vc, uint32_t src_id,
 	else if (vc == nae_fb_vc && size == 1)
 		process_tx_complete(cpu, src_id, msg0);
 	else {
-		printk("%s , Error invalid message, vc %d size %d\n",
+		nae_print(NAE_DBG_ERROR, "%s , Error invalid message, vc %d size %d\n",
 			__func__, vc, size);
 	}
 }
@@ -1023,40 +1023,40 @@ extern int nlm_xlp_register_napi_final_handler(int major,
 int nlm_xlp_enable_napi(void)
 {
 	if (exclusive_vc) {
-		printk("Registering exclusive napi vc handler....\n");
+		nae_print(NAE_DBG_TRACE, "Registering exclusive napi vc handler....\n");
 		nlm_xlp_register_napi_vc_handler(nae_rx_vc, xlp_nae_napi_poll);
 		nlm_xlp_register_napi_vc_handler(nae_fb_vc, xlp_nae_napi_poll);
 
 		return 0;
 	}
 
-	printk("Registering  poe0->cpu msgring handler\n");
+	nae_print(NAE_DBG_TRACE, "Registering  poe0->cpu msgring handler\n");
 	if (register_xlp_msgring_handler(XLP_MSG_HANDLE_POE_0,
 		nlm_xlp_nae_msgring_handler, NULL)) {
-		printk("Fatal error! Can't register msgring handler "
+		nae_print(NAE_DBG_ERROR, "Fatal error! Can't register msgring handler "
 		       "for XLP_MSG_HANDLE_NAE_0\n");
 		return -1;
 	}
-	printk("Registering nae-0 FB msgring handler\n");
+	nae_print(NAE_DBG_TRACE, "Registering nae-0 FB msgring handler\n");
 	if (register_xlp_msgring_handler(XLP_MSG_HANDLE_NAE_0,
 		nlm_xlp_nae_msgring_handler, NULL)) {
-		printk("Fatal error! Can't register msgring handler "
+		nae_print(NAE_DBG_ERROR, "Fatal error! Can't register msgring handler "
 		       "for XLP_MSG_HANDLE_NAE_0\n");
 		return -1;
 	}
 
 	if(is_nlm_xlp9xx()){
-		printk("Registering  poe1->cpu msgring handler\n");
+		nae_print(NAE_DBG_TRACE, "Registering  poe1->cpu msgring handler\n");
 		if (register_xlp_msgring_handler(XLP_MSG_HANDLE_POE_1,
 			nlm_xlp_nae_msgring_handler, NULL)) {
-			printk("Fatal error! Can't register msgring handler "
+			nae_print(NAE_DBG_ERROR, "Fatal error! Can't register msgring handler "
 			       "for XLP_MSG_HANDLE_NAE_0\n");
 			return -1;
 		}
-		printk("Registering nae-1 FB msgring handler\n");
+		nae_print(NAE_DBG_TRACE, "Registering nae-1 FB msgring handler\n");
 		if (register_xlp_msgring_handler(XLP_MSG_HANDLE_NAE_1,
 			nlm_xlp_nae_msgring_handler, NULL)) {
-			printk("Fatal error! Can't register msgring handler "
+			nae_print(NAE_DBG_ERROR, "Fatal error! Can't register msgring handler "
 		       	"for XLP_MSG_HANDLE_NAE_0\n");
 			return -1;
 		}
@@ -1064,12 +1064,12 @@ int nlm_xlp_enable_napi(void)
 
 #ifdef CONFIG_INET_LRO
 	if (enable_lro) {
-		printk("Registering poe0->cpu napi final handler\n");
+		nae_print(NAE_DBG_TRACE, "Registering poe0->cpu napi final handler\n");
 		nlm_xlp_register_napi_final_handler(XLP_MSG_HANDLE_POE_0,
 				xlp_napi_lro_flush, NULL);
 
 		if(is_nlm_xlp9xx()){
-			printk("Registering poe1->cpu napi final handler\n");
+			nae_print(NAE_DBG_TRACE, "Registering poe1->cpu napi final handler\n");
 			nlm_xlp_register_napi_final_handler(XLP_MSG_HANDLE_POE_1,
 					xlp_napi_lro_flush, NULL);
 		}
diff --git a/drivers/net/ethernet/broadcom/nae/xlpge_tso.h b/drivers/net/ethernet/broadcom/nae/xlpge_tso.h
index 8d1ce04..8a9bdcf 100644
--- a/drivers/net/ethernet/broadcom/nae/xlpge_tso.h
+++ b/drivers/net/ethernet/broadcom/nae/xlpge_tso.h
@@ -163,7 +163,7 @@ static inline int tso_xmit_skb(struct sk_buff *skb, struct net_device *dev)
 	uint32_t pad_len = 0, icv_len = 0, param_index = 0;
 
 #ifdef MACSEC_DEBUG
-	printk("nae_cfg->sectag_offset = %d sectag_len = %d icv_len = %d\n",
+	nae_print(NAE_DBG_TRACE, "nae_cfg->sectag_offset = %d sectag_len = %d icv_len = %d\n",
 		nae_cfg->sectag_offset[priv->port],
 		nae_cfg->sectag_len[priv->port], nae_cfg->icv_len[priv->port]);
 #endif
@@ -174,7 +174,7 @@ static inline int tso_xmit_skb(struct sk_buff *skb, struct net_device *dev)
 
 #ifdef MACSEC_DEBUG
 	dump_buffer(skb->data, skb->len, "Org skb pkt:");
-	printk("msec_port = %x priv->port = %d priv->block = %d \
+	nae_print(NAE_DBG_TRACE, "msec_port = %x priv->port = %d priv->block = %d \
 		priv->index = %d skb->len = %d \
 		nae_cfg->msec_port_enable = %x\n",
 		msec_port, priv->port, priv->block, priv->index,
@@ -188,7 +188,7 @@ static inline int tso_xmit_skb(struct sk_buff *skb, struct net_device *dev)
 					MAC_HEADER_LEN);
 
 #ifdef MACSEC_DEBUG
-	printk("skb->len = %d ether_type = %x\n",
+	nae_print(NAE_DBG_TRACE, "skb->len = %d ether_type = %x\n",
 			skb->len, ether_type);
 #endif
 		/* Enable Macsec processing */
@@ -202,7 +202,7 @@ static inline int tso_xmit_skb(struct sk_buff *skb, struct net_device *dev)
 			icv_len = nae_cfg->icv_len[priv->port];
 
 #ifdef MACSEC_DEBUG
-	printk("pad_len = %d icv_len = %d ether_type = %x\n",
+	nae_print(NAE_DBG_TRACE, "pad_len = %d icv_len = %d ether_type = %x\n",
 			pad_len, icv_len, ether_type);
 #endif
 		}
@@ -240,7 +240,7 @@ static inline int tso_xmit_skb(struct sk_buff *skb, struct net_device *dev)
 			udp_hdr(skb)->check = 0;
 			tcp_packet = 0;
 		} else {
-			printk("Invalid packet in %s\n", __FUNCTION__);
+			nae_print(NAE_DBG_ERROR, "Invalid packet in %s\n", __FUNCTION__);
 			goto out_unlock;
 		}
 
@@ -282,7 +282,7 @@ static inline int tso_xmit_skb(struct sk_buff *skb, struct net_device *dev)
 #ifdef MACSEC_DEBUG
 			dump_buffer((char *)&p2pdesc[P2P_SKB_OFF-8],
 				pad_len, "first_seg:");
-			printk(" len = %d pad_len = %d icv_len = %d \
+			nae_print(NAE_DBG_TRACE, " len = %d pad_len = %d icv_len = %d \
 				param_index = %d\n", len, pad_len, icv_len,
 				param_index);
 #endif
@@ -359,7 +359,7 @@ static int tso_enable(struct net_device *dev, u32 data)
 	if(!(is_nlm_xlp2xx_b0() || is_nlm_xlp9xx())) {
 		/* If force tso is not set, enable gso in that case */
 		if(!force_tso) {
-			printk("Disabling TSO & Enabling GSO \n");
+			nae_print(NAE_DBG_TRACE, "Disabling TSO & Enabling GSO \n");
 			dev->features = (dev->features & ~NETIF_F_TSO) | NETIF_F_GSO;
 			dev->hw_features = (dev->hw_features & ~NETIF_F_TSO) | NETIF_F_GSO;
 		}
diff --git a/drivers/net/ethernet/broadcom/nae/xlpge_tx.c b/drivers/net/ethernet/broadcom/nae/xlpge_tx.c
index cd2beb7..4c03194 100644
--- a/drivers/net/ethernet/broadcom/nae/xlpge_tx.c
+++ b/drivers/net/ethernet/broadcom/nae/xlpge_tx.c
@@ -73,7 +73,7 @@ int nlm_xlp_nae_start_xmit(struct sk_buff *skb, struct net_device *dev)
 	nae_t* nae_cfg = priv->nae;
 
 #ifdef MACSEC_DEBUG
-	printk("nae_cfg->sectag_offset = %d sectag_len = %d icv_len = %d\n",
+	nae_print(NAE_DBG_TRACE, "nae_cfg->sectag_offset = %d sectag_len = %d icv_len = %d\n",
 		nae_cfg->sectag_offset[priv->port],
 		nae_cfg->sectag_len[priv->port], nae_cfg->icv_len[priv->port]);
 #endif
@@ -84,7 +84,7 @@ int nlm_xlp_nae_start_xmit(struct sk_buff *skb, struct net_device *dev)
 
 #ifdef MACSEC_DEBUG
 	dump_buffer(skb->data, skb->len, "Org skb pkt:");
-	printk("msec_port = %x priv->port = %d priv->block = %d \
+	nae_print(NAE_DBG_TRACE, "msec_port = %x priv->port = %d priv->block = %d \
 		priv->index = %d skb->len = %d \
 		nae_cfg->msec_tx_port_enable = %x\n",
 		msec_port, priv->port, priv->block, priv->index,
@@ -98,7 +98,7 @@ int nlm_xlp_nae_start_xmit(struct sk_buff *skb, struct net_device *dev)
 #ifdef MACSEC_DEBUG
 	short ether_type = *(short*)(((char*)skb->data) +
 				MAC_HEADER_LEN);
-	printk("skb->len = %d ether_type = %x\n",
+	nae_print(NAE_DBG_TRACE, "skb->len = %d ether_type = %x\n",
 			skb->len, ether_type);
 #endif
 		/* Enable Macsec processing */
@@ -113,7 +113,7 @@ int nlm_xlp_nae_start_xmit(struct sk_buff *skb, struct net_device *dev)
 			icv_len = nae_cfg->icv_len[priv->port];
 
 #ifdef MACSEC_DEBUG
-	printk("pad_len = %d icv_len = %d ether_type = %x\n",
+	nae_print(NAE_DBG_TRACE, "pad_len = %d icv_len = %d ether_type = %x\n",
 			pad_len, icv_len, ether_type);
 #endif
 		}
@@ -123,11 +123,11 @@ int nlm_xlp_nae_start_xmit(struct sk_buff *skb, struct net_device *dev)
 
 #ifdef ENABLE_SANITY_CHECKS
 	if (!skb) {
-		printk("[%s] skb is NULL\n",__FUNCTION__);
+		nae_print(NAE_DBG_ERROR, "[%s] skb is NULL\n",__FUNCTION__);
 		return -1;
 	}
 	if (skb->len == 0) {
-		printk("[%s] skb empty packet\n",__FUNCTION__);
+		nae_print(NAE_DBG_ERROR, "[%s] skb empty packet\n",__FUNCTION__);
 		return -1;
 	}
 #endif
-- 
1.7.1

